### ISSUE TITLE
Remove the reportImplicitOverride error when working with abstract classes or protocols.

### ISSUE THREAD

**Issue by vlkorsakov** [DetachHead/basedpyright#986]
I think it’s a false positive. Basedpyright, like pyright, shows this error when I implement an abstract class or protocol:
`Method "x" is not marked as override but is overriding a method in class "Interface"basedpyright[reportImplicitOverride](https://docs.basedpyright.com/v1.23.2/configuration/config-files/#reportImplicitOverride)`

```python
class Interface(Protocol):
    def x(self) -> None:
        raise NotImplementedError


class InterfaceABC(ABC):
    @abstractmethod
    def x(self) -> None:
        raise NotImplementedError


class Implementation(Interface):
    def x(self) -> None:
        ...


class ImplementationABC(Interface):
    def x(self) -> None:
        ...
```

Maybe, when I implement a protocol (abstract class) method or a method with the `@abstractmethod` decorator, basedpyright shouldn't show this error.

**Comment by DetachHead** [comment]
`reportImplicitOverride` is good at catching mistakes when renaming or removing methods. for example, if the `x` method was removed from `InterfaceABC`, you'd still want an error on `ImplementationABC.x` informing you that the method you're overriding/implementing no longer exists, which means it would probably never get called. so i think that rule should still be reported in cases like this

**Comment by DetachHead** [comment]
closing for the reason mentioned in my previous comment. if you disagree feel free to comment or raise another issue and we can discuss further

**Comment by Velocidensity** [comment]
I'm not sure I agree with the argument here. Having the warning show up, if the method _is_ removed would be good, but _only_ then, not at the cost of having it on every abstractmethod, as it currently seems to be.

**Comment by DetachHead** [comment]
if i add the option to disable this diagnostic for abstract methods only, it would either need to be split into two separate diagnostic rules, or a separate option would need to be created which changes the behavior of this rule. to maintain backwards compatibility with pyright, the latter is probably the preferred option. though i still don't really understand why you would want abstract methods to be treated differently by this rule.
